clippy_lints/src/ignored_unit.rs

```diff
@@ -0,0 +1,54 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use hir::PatKind;
+use rustc_errors::Applicability;
+use rustc_hir as hir;
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for usage of `_` in patterns of type `()`.
+    ///
+    /// ### Why is this bad?
+    /// Matching with `()` explicitely instead of `_` outlines
```

Comment:
```suggestion
    /// Matching with `()` explicitly instead of `_` outlines
```

Reply:
This should probably also mention that it prevents the type from changing as well. Usually a good thing, other times not.